WordPress/openverse-frontend
# Move PO file into special branch

## Fixes


Fixes #1051 by @dhruvkb

## Description


Fixes the problem described by @dhruvkb in the linked issue by completely changing how we manage the POT file altogether :slightly_smiling_face: 

You can check out the new `translations` branch here: https://github.com/WordPress/openverse-frontend/tree/translations

The new `generate_pot.yml` workflow will push a new version of the `openverse.pot` file every time we merge to main.

We'll need to update the GlotPress configuration to pull `openverse.pot` from the translations branch specifically once this is merged. cc @zackkrida if you can document how that is done that would be great!

Because the PO file is just used to hydrate GlotPress and then we generate the json locale files from the GlotPress API responses, we don't need to store the PO file in the main branch at all. Everything is derived.

## Testing Instructions

Verify the approach is sane and will work. For the workflow, check that the code makes sense. Then we'll find out if it works when we merge it.

## Checklist

- [N/A] I added or updated tests for the changes I made (if applicable).
- [N/A] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
